### PR TITLE
enqueue_file_kwargs to fix SRS client

### PR DESCRIPTION
**Pull request by nabobalis**

### CODE REVIEW

File: sunpy/net/dataretriever/sources/noaa.py [hunk 1]
```diff
@@ -162,14 +162,7 @@ class SRSClient(GenericClient):
...
     pattern = '{}/{year:4d}/{month:2d}/{year:4d}{month:2d}{day:2d}SRS.txt'
-
-    def fetch(self, *args, **kwargs):
-        """
-        Download a set of results.
-        """
-        # Server does not support the normal aioftp passive command.
-        kwargs["passive_commands"] = ["pasv"]
-        super().fetch(*args, **kwargs)
+    enqueue_file_kwargs = {"passive_commands": "pasv"}
```
Comment (Cadair) [inline sunpy/net/dataretriever/sources/noaa.py:165]: Needs to be a list.

```suggestion
    enqueue_file_kwargs = {"passive_commands": ["pasv"]}
```

Comment (nabobalis) [reply sunpy/net/dataretriever/sources/noaa.py]: Can't believe I glossed over that.

File: sunpy/net/dataretriever/sources/noaa.py [hunk 2]
```diff
@@ -162,14 +162,7 @@ class SRSClient(GenericClient):
...
     pattern = '{}/{year:4d}/{month:2d}/{year:4d}{month:2d}{day:2d}SRS.txt'
-
-    def fetch(self, *args, **kwargs):
-        """
-        Download a set of results.
-        """
-        # Server does not support the normal aioftp passive command.
-        kwargs["passive_commands"] = ["pasv"]
-        super().fetch(*args, **kwargs)
+    enqueue_file_kwargs = {"passive_commands": ["pasv"]}
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/noaa.py:165]:
```suggestion
    # Server does not support the normal aioftp passive command.
    enqueue_file_kwargs = {"passive_commands": ["pasv"]}
```

**Comment by nabobalis** [comment]
```
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/hek2vso/tests/test_hek2vso.py::test_full_query
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/hek2vso/tests/test_hek2vso.py::test_quick_clean
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/tests/test_fido.py::test_vso_fetch_hmi
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/vso/tests/test_vso.py::test_path
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/vso/tests/test_vso.py::test_vso_hmi
FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/vso/tests/test_vso.py::test_response_block_properties
FAILED ../../docs/guide/acquiring_data/fido.rst::fido.rst
FAILED ../../docs/guide/acquiring_data/hek.rst::hek.rst
```
SRS tests are passing again.

**Review by dstansby — CHANGES_REQUESTED** [review]
This should definitely get a bugfix changelog entry.

**Comment by Cadair** [comment]
Everything but CI fixes that is backported should get a changelog entry.

**Comment by Cadair** [comment]
Also I am going to want to see the online test suite go green.

**Comment by nabobalis** [comment]
Based on the online fails and the fact that main cron job only had the SRS tests fail. These extra ones are probably legit.

File: sunpy/net/dataretriever/client.py [hunk 3]
```diff
@@ -64,6 +64,8 @@ class GenericClient(BaseClient):
     pattern = None
     # Set of required 'attrs' for client to handle the query.
     required = {a.Time, a.Instrument}
+    # Define keywards a client needs to pass to enqueue_file
+    enqueue_file_kwargs = {}
```
Comment (Cadair) [inline sunpy/net/dataretriever/client.py:68]: We should probably document this new feature somewhere. Maybe the Fido dev docs?

**Review by Cadair — APPROVED** [review]

File: sunpy/net/dataretriever/client.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/net/dataretriever/client.py:68]: Can do.

**Review by dstansby — APPROVED** [review]

File: sunpy/net/dataretriever/client.py (hunk 3 above)
Comment (Cadair) [reply sunpy/net/dataretriever/client.py:68]: Should we open an issue for this?

Comment (nabobalis) [reply sunpy/net/dataretriever/client.py:68]: I forgot to do this change before it was merged. I will open a PR now.

Comment (nabobalis) [reply sunpy/net/dataretriever/client.py:68]: THE TYPO TOO

**Pull request merged**